### ISSUE TITLE
(rpm)-ostree pkg: prepare for jigdo 

### ISSUE THREAD

**Issue by cgwalters** [cockpit-project/cockpit#8435]
We are in the process of a fundamental rework to rpm-ostree to add [jigdo mode ♲📦](https://github.com/projectatomic/rpm-ostree/issues/1081).

Since the current `ostree.js` pkg does things like parse refspecs, it's going to have to learn about this.  We're still debating the design on the DBus side though.

It'd actually be very useful to think about how this looks and feels on the Cockpit side.  Note this also strongly intersects with our push for [automatic updates 🆕⌚️](https://github.com/projectatomic/rpm-ostree/issues/247).

One thing I want to specifically highlight is [this subthread](https://github.com/projectatomic/rpm-ostree/issues/1081#issuecomment-348540604).

Basically the analogies we're currently building are:

 - ostree remote = rpm-md (yum) repo
 - ostree refspec = jigdoRPM package name + (enabled repo state, just like packages)
 - ostree version = jigdoRPM version

For example, after jigdo there's no separate `fedora/27/x86_64/atomic-host` and `fedora/27/x86_64/updates-testing/atomic-host`.  A bit like traditional, you'd `rpm-ostree rebase --repo=updates-testing` which would be short for going from `fedora:fedora-atomic-host` to `fedora-updates-testing:fedora-atomic-host`.

**Comment by martinpitt** [comment]
Thanks @cgwalters for the early warning. This is really cool work BTW, and it's awesome that it simplifies the server side so much!

In cockpit we need to support both the traditional and jigdo ostree for a while, so I figure we need some dynamic API tests. Is there already some early Atomic image available which uses jigdo, or some COPR or jigdo-ostree.rpm that can be installed into an existing Atomic image, to play around with this? Even though the D-Bus API might not be final yet, it would already allow us to get more familiar with the new concepts and look-and-feel.

Will the switch to ostree-jigdo be possible on upgrades to existing Atomics? I figure the current ostree remote+refspec might be sitting in /etc and are not part of the ostree itself; can the upgrade process somehow adjust them? I figure once this goes life, we don't want to keep the old ostree repos around for a long time?
